### PR TITLE
FastFit implemented

### PR DESCRIPTION
**Pull request by peressounko**
Fitting with semigaus finctions re-implemented

### CODE REVIEW

**Comment by peressounko** [comment]
Dear @shahor02 , could you please have a look at this PR? New fitter of raw samples is implemented which should replace default one. We need to test it in FLPs and in another pass of Pilot run reconstruction.

File: Detectors/PHOS/reconstruction/src/CaloRawFitterGS.cxx [hunk 1]
```diff
@@ -108,208 +119,172 @@ CaloRawFitterGS::FitStatus CaloRawFitterGS::evalFit(gsl::span<short unsigned int
...
+  // Find zero of 4-order polinomial
+  // first use linear extrapolation to reach correct root of four
+  double z = -1.;
+  if (ma0[j] * b1 - ma1[j] * b0 != 0) {
+    z = (ma1[j] * b1 - ma2[j] * b0) / (ma0[j] * b1 - ma1[j] * b0) - 1.; // linear fit + offset
+  }
+  double q = 0., dq = 0., ddq = 0., lq = 0., dz = 0.1;
+  q = a * z * z * z * z + b * z * z * z + c * z * z + d * z + e; // polinomial
+  dq = 4. * a * z * z * z + 3. * b * z * z + 2. * c * z + d;     // Derivative
+  ddq = 12. * a * z * z + 6. * b * z + 2. * c;                   // Second derivative
```
Comment (shahor02) [inline Detectors/PHOS/reconstruction/src/CaloRawFitterGS.cxx]: you could decrease the number of multiplications by precalculated quantities.

**Review by shahor02 — DISMISSED** [review]
Hi @peressounko 
looks fine, though I did not enter into the math of the fit. A tiny comment below, up to you if you want to implement it now or merge it as it is.

File: Detectors/PHOS/reconstruction/src/CaloRawFitterGS.cxx (hunk 1 above)
Comment (peressounko) [reply Detectors/PHOS/reconstruction/src/CaloRawFitterGS.cxx]: Hi @shahor02, thank you, I modified code. As for math - I checked, method reproduces fitting with Minuit with good accuracy.

**Comment by peressounko** [comment]
@shahor02 now code is ready to merge, could you approve it?

**Review by shahor02 — APPROVED** [review]

**Pull request merged**